src/serialization.jl

```diff
@@ -0,0 +1,72 @@
+import JSON
+
+const SUPPORTED_TYPES = Union{EventValues, EventStats}
+
+JSON.lower(es::Event) = event_to_name(es)
+function JSON.lower(x::Union{EventValues, EventStats})
+    d = Dict{String,Any}()
+    T = typeof(x)
+    for i = 1:nfields(x)
+        name = String(fieldname(T, i))
+        value = getfield(x, i)
+        ft = typeof(value)
+        d[name] = value
+    end
+
+    [string(T.name.name), d]
+end
+
+function recover(x::Vector)
+    length(x) == 2 || throw(ArgumentError("Expecting a vector of length 2"))
+    typename = x[1]::String
+    fields = x[2]::Dict
+
+    T = if typename == "EventValues"
+        EventValues
+    elseif typename == "EventStats"
+        EventStats
+    else
+        throw(ArgumentError("Unknown typename $typename"))
+    end
+
+    fc = fieldcount(T)
+    xs = map(1:fc) do i
+        ft = fieldtype(T, i)
+        fn = String(fieldname(T, i))
+
+        if ft == Vector{Event}
```

Comment:
same kind of ugly. should implement the inverse of JSON.lower
but only for 2 types for now